src/pharmpy/model/statements.py

```diff
@@ -248,17 +249,17 @@ class ExplicitODESystem(ODESystem):
     ⎩A_CENTRAL(0) = 0
     """
 
-    def __init__(self, odes: List[sympy.Eq], ics: Dict[sympy.Expr, sympy.Expr]):
+    def __init__(self, odes: Tuple[sympy.Eq, ...], ics: Mapping[sympy.Expr, sympy.Expr]):
         self._odes = odes
-        self._ics = ics
+        self._ics = MappingProxyType(ics)  # NOTE This guarantees immutability
```

Comment:
We could also do a copy of the input dictionary before making it a read-only view if necessary.

Reply:
Copying is needed since changing the underlying `dict` still changes the `MappingProxyType` object. Ok, I thought `frozendict` was in the standard library.

Reply:
OK, I'll add a copy.

Reply:
Done with bb3d88694e97af0fedeb2c90392861773e191f04